### PR TITLE
WYSIWYG editor delegation a bit smarter

### PR DESCRIPTION
**Pull request by mkarpicz**

### CODE REVIEW

File: static/js/fit.js [hunk 1]
```diff
@@ -981,24 +981,11 @@ $( document ).ready(function() {
...
-  });
-
-  delegate('#add-edit-resource', 'submit', (e,t) => {
-    let mysave = $('#fit_wysiwyg_resource').html();
-    $('#resource_description').val(mysave);
+  delegate('[data-fit-wysiwyg]', 'submit', (e,t) => {
+    let wysiwygId = t.dataset.fitWysiwygId;
+    let mysave = $(`[data-fit-wysiwyg-div-${wysiwygId}]`).html();
+    let testarea = $(`[data-textarea-wysiwyg-${wysiwygId}]`);
+    testarea.val(mysave);
```
Comment (Yuffster) [inline static/js/fit.js]: Very nice work, @mkarpicz!

You can put the ID directly into the `data-fit-wysiwyg` property to define your target element and not need the `data-fit-wysiwyg-id` attribute.  This is how a lot of Bootstrap stuff works. :)

Comment (Yuffster) [reply static/js/fit.js]: The dynamic attributes are a bit strange, why not try using two data attributes to specify the preview and the input target?

Comment (Yuffster) [reply static/js/fit.js]: Or another way to do it...

```.html
<form data-fit-wysiwyg>
    <textarea name="course_summary" style="display: none" data-textarea-wysiwyg-input></textarea>
    <div data-fit-wysiwyg-preview></div>
</form>
```

```.js
delegate('[data-fit-wysiwyg]', 'submit', (e, t) => {
    let p = t.closest('[data-fit-wysiwyg]');
    let mysave = p.querySelector('[data-fit-wysiwyg-input]').html();
    let testarea = p.querySelector('[data-fit-wysiwyg-preview]').html();
    /* ... */
});
```

File: templates/course_edit.html [hunk 2]
```diff
@@ -120,7 +120,7 @@ <h3>
           <h3>
             <span class="txt">Course info.</span>
           </h3>
-          <form action="/course/{{ course.slug }}/edit" method="post" id="course-edit-form">
+          <form action="/course/{{ course.slug }}/edit" method="post" id="course-edit-form" data-fit-wysiwyg data-fit-wysiwyg-id="course_summary">
```
Comment (Yuffster) [inline templates/course_edit.html]: What are you using `data-fit-wysiwyg-id` for?

File: static/js/fit.js [hunk 3]
```diff
@@ -981,24 +981,12 @@ $( document ).ready(function() {
...
-  });
-
-  delegate('#course-edit-form', 'submit', (e,t) => {
-    let mysave = $('#fit_wysiwyg_editor').html();
-    $('#course_summary').val(mysave);
-  });
-
-  delegate('#add-edit-resource', 'submit', (e,t) => {
-    let mysave = $('#fit_wysiwyg_resource').html();
-    $('#resource_description').val(mysave);
```
Comment (Yuffster) [inline static/js/fit.js]: Very nice! 👨‍🍳

File: templates/course_edit.html [hunk 4]
```diff
@@ -178,9 +178,9 @@ <h3>
               <dd>
                 <div class="form-group">
                   <div class="fit_wysiwyg">
-                    <div id="fit_wysiwyg_editor">{% if course.summary_html %}{{ course.summary_html|safe }}{% endif %}</div>
+                    <div id="fit_wysiwyg_editor" data-fit-wysiwyg-preview>{% if course.summary_html %}{{ course.summary_html|safe }}{% endif %}</div>
                   </div>
-                  <textarea id="course_summary" name="course_summary" style="display: none"></textarea>
+                  <textarea id="course_summary" name="course_summary" style="display: none" data-fit-wysiwyg-textarea></textarea>
```
Comment (Yuffster) [inline templates/course_edit.html]: If there are no other text areas within `data-fit-wysiwyg` divs, you can just search for `p.querySelector('textarea')` and not need this data attribute.

File: static/js/fit.js [hunk 5]
```diff
@@ -981,24 +981,12 @@ $( document ).ready(function() {
...
+  delegate('[data-fit-wysiwyg]', 'submit', (e,t) => {
+    let p = t.closest('[data-fit-wysiwyg]');
+    let preview = p.querySelector('[data-fit-wysiwyg-preview]');
+    let testarea = p.querySelector('[data-fit-wysiwyg-textarea]');
```
Comment (Yuffster) [inline static/js/fit.js]: You should change this variable to `textarea` to avoid confusion. 😄

**Comment by Yuffster** [comment]
#153 #164

File: templates/course_edit.html (hunk 2 above)
Comment (mkarpicz) [reply templates/course_edit.html]: Missed that one.

File: templates/course_edit.html (hunk 4 above)
Comment (mkarpicz) [reply templates/course_edit.html]: Indeed, that's no point in duplicating stuff.

File: static/js/fit.js (hunk 5 above)
Comment (mkarpicz) [reply static/js/fit.js]: 😅

**Pull request merged**